src/ProjectTemplates/Microsoft.Extensions.AI.Templates/src/ChatWithCustomData/ChatWithCustomData.Web/README.md

```diff
@@ -0,0 +1,209 @@
+    
+   ```sh
+   cd <<your-project-directory>>
+   dotnet user-secrets set AzureOpenAI:Key YOUR-AZURE-OPENAI-KEY
+   dotnet user-secrets set AzureOpenAI:Endpoint YOUR-AZURE-OPENAI-ENDPOINT
+   ```
+#### ---#endif
+
+Make sure to replace `YOUR-AZURE-OPENAI-KEY` and `YOUR-AZURE-OPENAI-ENDPOINT` with your actual Azure OpenAI key and endpoint.
+
+#### ---#endif
+#### ---#endif
+#### ---#if (UseAzureAISearch)
+## Configure Azure AI Search
+
+To use Azure AI Search, you will need an Azure account and an Azure AI Search resource. For detailed instructions, see the [Azure AI Search documentation](https://learn.microsoft.com/azure/search/search-create-service-portal).
+
+### 1. Create an Azure AI Search Resource
+Follow the instructions in the [Azure portal](https://portal.azure.com/) to create an Azure AI Search resource.
+
+#### ---#if (UseManagedIdentity)
+### 2. Configure Azure AI Search for Keyless Authentication
+This template is configured to use keyless authentication (also known as Managed Identity, with Entra ID). Before continuing, you'll need to configure your Azure AI Search resource to support this. [Learn more](https://learn.microsoft.com/azure/search/keyless-connections).
+
+### 3. Set the Azure AI Search Endpoint for this app
+   Configure your Azure AI Search endpoint for this project, using .NET User Secrets:
+   1.  In the Azure Portal, navigate to your Azure AI Search resource.
+   2.  Copy the "URL" from the "Overview" section.
+#### ---#if (hostIdentifier == "vs")
+   3.  In Visual Studio, right-click on your project in the Solution Explorer and select "Manage User Secrets".
+   4.  This will open a `secrets.json` file where you can store your Azure AI Search endpoint securely. Add the following key & value to the file:
+
+   ```json
+   {
+      "AzureAISearch:Endpoint": "YOUR-AZURE-AI-SEARCH-ENDPOINT"
+   }
+   ```
+#### ---#else
+   3.  From the command line, configure your Azure AI Search endpoint for this project using .NET User Secrets by running the following commands:
+
+   ```sh
+   cd <<your-project-directory>>
+   dotnet user-secrets set AzureAISearch:Endpoint YOUR-AZURE-AI-SEARCH-ENDPOINT
+   ```
+#### ---#endif
+
+   Make sure to replace `YOUR-AZURE-AI-SEARCH-ENDPOINT` with your actual Azure AI Search endpoint.
+
+#### ---#else
+### 3. Configure API Key and Endpoint
+   Configure your Azure AI Search API key and endpoint for this project, using .NET User Secrets:
+   1.  In the Azure Portal, navigate to your Azure AI Search resource.
+   2.  Copy the "Endpoint" URL and "Primary admin key" from the "Keys" section.
+#### ---#if (hostIdentifier == "vs")
+   3. In Visual Studio, right-click on your project in the Solution Explorer and select "Manage User Secrets".
+   4. This will open a `secrets.json` file where you can store your API key and endpoint securely, add the following keys & values to the file:
```

Comment:
```suggestion
   4. This will open a `secrets.json` file where you can store your API key and endpoint without them being tracked in source control. Add the following keys and values to the file:
```